ReactiveCocoaTests/Swift/SignalSpec.swift

```diff
@@ -46,12 +170,66 @@ class SignalSpec: QuickSpec {
 			pending("should deallocate after completing") {
 			}
 
-			pending("should forward events to observers") {
+			it("should forward events to observers") {
+				let (signal, observer) = Signal<Int, NoError>.pipe()
+				
+				var fromSignal: [Int] = []
+				var completed = false
+				
+				signal.observe(next: { number in
+					fromSignal.append(number)
+				}, completed: {
+					completed = true
+				})
+				
+				expect(fromSignal).to(beEmpty())
+				expect(completed).to(beFalsy())
+				
+				sendNext(observer, 1)
+				expect(fromSignal).to(equal([1]))
+				
+				sendNext(observer, 2)
+				expect(fromSignal).to(equal([1, 2]))
+				
+				expect(completed).to(beFalsy())
+				sendCompleted(observer)
+				expect(completed).to(beTruthy())
 			}
 		}
 
 		describe("observe") {
-			pending("should stop forwarding events when disposed") {
+			var testScheduler: TestScheduler!
+			
+			beforeEach {
+				testScheduler = TestScheduler()
+			}
+			
+			it("should stop forwarding events when disposed") {
+				let disposable = SimpleDisposable()
+				
+				let signal: Signal<Int, NoError> = Signal { observer in
+					testScheduler.schedule {
+						for number in [1, 2] {
+							sendNext(observer, number)
+						}
+						sendCompleted(observer)
+						sendNext(observer, 4)
+					}
+					return disposable
+				}
+				
+				var fromSignal: [Int] = []
+				signal.observe(next: { number in
+					fromSignal.append(number)
+				})
+				
+				expect(disposable.disposed).to(beFalsy())
+				expect(fromSignal).to(beEmpty())
+				
+				testScheduler.run()
+				
+				expect(disposable.disposed).to(beTruthy())
+				expect(fromSignal).to(equal([1, 2]))
 			}
 
 			pending("should not trigger side effects") {
```

Comment:
If you put some side effects (like incrementing a counter) in the generator passed to `init`, multiple `observe` invocations shouldn't affect how many times those side effects occur.

In other words, `Signal` effects should always occur _exactly_ once, and observers don't change that.